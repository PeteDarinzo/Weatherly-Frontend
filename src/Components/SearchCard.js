import React from "react";
import { useDispatch } from "react-redux";
import Card from '@mui/material/Card';
import CardActions from '@mui/material/CardActions';
import CardContent from '@mui/material/CardContent';
import CardMedia from '@mui/material/CardMedia';
import Button from '@mui/material/Button';
import Typography from '@mui/material/Typography';
import { sendMovieToAPI } from "../Actions/actions";

// const SearchCard = ({ id, title, posterUrl }) => {
const SearchCard = ({ movie, saveMovie }) => {

  const dispatch = useDispatch();

  function handleClick() {
    saveMovie(movie);
    // dispatch(sendMovieToAPI(movie));
  }

  // function handleClick() {
  //   dispatch(sendMovieToAPI(1, id, title, posterUrl));
  // }

  return (
    <Card sx={{ maxWidth: 345 }}>
      <CardMedia
        component="img"
        height="140"
        image={movie.Poster}
        alt={movie.Title}
        sx={{ maxWidth: "50%" }}
      />
      <CardContent>
        <Typography gutterBottom variant="h5" component="div">
          {movie.Title}
        </Typography>
      </CardContent>
      <CardActions>
        <Button onClick={handleClick} size="small">Save</Button>
      </CardActions>
    </Card>
  );
}

export default SearchCard;