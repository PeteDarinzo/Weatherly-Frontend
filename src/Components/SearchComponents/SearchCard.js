import React, { useState } from "react";
import Card from '@mui/material/Card';
import CardActions from '@mui/material/CardActions';
import CardContent from '@mui/material/CardContent';
import CardMedia from '@mui/material/CardMedia';
import Button from '@mui/material/Button';
import Typography from '@mui/material/Typography';

const SearchCard = ({ movie, saveMovie }) => {

  function handleClick(e) {
      saveMovie(movie);
  }

  return (
    <Card sx={{ maxWidth: '275px', mx: "auto", my: 3, backgroundColor: "#2b2b2b" }} >
      <CardMedia
        component="img"
        image={movie.Poster}
        alt={movie.Title}
        sx={{ width: '100%', height: '100%' }}
      />
      <CardContent>
        <Typography variant="h6">
          {movie.Title}
        </Typography>
        <Typography>
          {movie.Year}
        </Typography>
      </CardContent>
      <CardActions>
        <Button onClick={handleClick} variant="contained">Save</Button>
      </CardActions>
    </Card>
  );
}

export default SearchCard;