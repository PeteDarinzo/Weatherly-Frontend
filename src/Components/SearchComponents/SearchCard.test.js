import SearchCard from "./SearchCard";
import { render } from '@testing-library/react';
import { MemoryRouter } from "react-router-dom";

it("renders a search card without crashing", () => {
  render(
    <MemoryRouter>
      <SearchCard movie={{ Title: "test", Year: "1999", Poster: "image_url" }} saveMovie={() => { }} />
    </MemoryRouter>
  );
});


it("matches the search card snapshot", () => {
  const { asFragment } = render(
    <MemoryRouter>
      <SearchCard movie={{ Title: "test", Year: "1999", Poster: "image_url" }} saveMovie={() => { }} />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
