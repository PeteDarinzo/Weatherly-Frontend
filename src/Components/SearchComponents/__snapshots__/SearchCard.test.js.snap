// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the search card snapshot 1`] = `
<DocumentFragment>
  <div
    class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-1gdl5pu-MuiPaper-root-MuiCard-root"
  >
    <img
      alt="test"
      class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-cljvgu-MuiCardMedia-root"
      src="image_url"
    />
    <div
      class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
      >
        test
      </h6>
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        1999
      </p>
    </div>
    <div
      class="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
    >
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Save
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</DocumentFragment>
`;
